docs/poc2023q1/outline.md

```diff
@@ -0,0 +1,431 @@
+# 2023q1 PoC
+
+## Introduction
+
+This is a quick demo of a fragment of what we think is needed for edge
+multi-cluster.  It is intended to demonstrate the following points.
+
+- Separation of inventory and workload management.
+- The focus here is on workload management, and that strictly reads
+  inventory.
+- What passes from inventory to workload management is kcp TMC
+  Location and SyncTarget objects.
+- Use of a kcp workspace as the container for the central spec of a workload.
+- Propagation of desired state from center to edge, as directed by
+  EdgePlacement objects and the Location and SyncTarget objects they reference.
+- Interfaces designed for a large number of edge clusters.
+- Interfaces designed with the intention that edge clusters operate
+  independently of each other and the center (e.g., can tolerate only
+  occasional connectivity) and thus any "service providers" (in the
+  technical sense from kcp) in the center or elsewhere.
+- Rule-based customization of desired state.
+- Propagation of reported state from edge to center.
+- Summarization of reported state in the center.
+- The edge opens connections to the center, not vice-versa.
+- An edge computing platform "product" that can be deployed (as
+  opposed to a service that is used).
+
+Some important things that are not attempted in this PoC include the following.
+
+- An implementation that supports a large number of edge clusters or
+  any other thing that requires sharding for scale.
+- More than one SyncTarget per Location.
+- Return or summarization of status from associated objects (e.g.,
+  ReplicaSet or Pod objects associated with a given Deployment
+  object).
+- A hierarchy with more than two levels.
+- User control over ordering of propagation from center to edge,
+  either among destinations or kinds of objects.
+- More than baseline security (baseline being, e.g., HTTPS, Secret
+  objects, non-rotating bearer token based service authentication).
+- A good design for bootstrapping the workload management in the edge
+  clusters.
+- Support for workload object types that are not either built into kcp
+  or imported via a kcp APIBinding.
+- Very strong isolation between tenants in the edge computing
+  platform.
+
+It is TBD whether the implementation will support intermittent
+connectivity.  This depends on whether we can quickly and easily get a
+syncer that creates the appropriately independent objects in the edge
+cluster and itself tolerates intermittent connectivity.
+
+As further matters of roadmapping development of this PoC:
+customization may be omitted at first, and summarization may start
+with only a limited subset of the implicit functionality.
+
+This PoC builds on TMC and makes some compromises to accommodate that.
+The implementation involves workload components (syncers) writing
+status information to inventory objects (SyncTargets).
```

Comment:
Help me understand what aspect of  that is a compromise - simply that it's incompatible with routinely/intermittently offline edge clusters?

Reply:
No, the compromise is the violation of the division into infrastructure and workload layers with the workload layer strictly consuming information from the infrastructure layer.  The SyncTarget objects come from the infrastructure layer, so it is not right for the workload layer (syncers) to update the SyncTarget objects.